src/plugins/proctracer/proctracer.cpp

```diff
@@ -0,0 +1,401 @@
+    ctx.dtb = info->regs->cr3;
+
+    addr_t process = drakvuf_get_current_process(drakvuf, info->vcpu, info->regs);
+
+    if (!process){ 
+        drakvuf_release_vmi(drakvuf);
+        return -1;
+    }
+
+    char *proc_name = drakvuf_get_process_name(drakvuf, process);
+    switch(p->format){
+        case OUTPUT_CSV:
+            printf("proctracer,start,%lx,\"%s\"\n", info->regs->cr3, proc_name);
+            break;
+        default:
+        case OUTPUT_DEFAULT:
+            printf("[PROCTRACER] Starting CR3: %lx NAME: %s\n", info->regs->cr3, proc_name);
+            break;
+    }
+    if (p->mod_config.find(proc_name) == p->mod_config.end() || p->trace_status.find(info->regs->cr3) != p->trace_status.end()){
+        free(proc_name);
+        drakvuf_release_vmi(drakvuf);
+        return 0;
+    }
+    add_trace_points(p, drakvuf, vmi, process, info->regs->cr3, proc_name);
+    free(proc_name);
+    drakvuf_release_vmi(drakvuf);
+    return 0;
+}
+
+proctracer::proctracer(drakvuf_t drakvuf, const void *config, output_format_t output) {
+    const struct proctracer_config *c = (const struct proctracer_config *)config;
+    const char *rekall_profile = c->rekall_profile;
+    const char *proctracer_config;
+    if (c->proctracer_config){
+        proctracer_config = c->proctracer_config;
+        printf("Proctracer config: %s\n",proctracer_config);
+    }else{
+        proctracer_config = "proctracer.json";    
+    }
+
+    printf("[PROCTRACER] Starting...\n");
+
+    this->offsets = (size_t*)g_malloc0(__OFFSET_MAX*sizeof(size_t));
+
+    for(int i=0;i<__OFFSET_MAX;i++) {
+        drakvuf_get_struct_member_rva(rekall_profile, offset_names[i][0], offset_names[i][1],
+                                      &this->offsets[i]);
+    }
+
+    json_object *conf_root = json_object_from_file(proctracer_config);
+    if (!conf_root){
+        printf("[PROCTRACER] Can't find config!\n");
+        throw -1;
+    }
+    printf("[PROCTRACER] Main config loaded\n");
+    json_object *conf_modules=NULL;
+    json_object *conf_ccov_limit=NULL;
+    if (!json_object_object_get_ex(conf_root, "modules", &conf_modules)) {
+        printf("[PROCTRACER] Can't find any modules to trace\n");
+        json_object_put(conf_root);
+        throw -1;
+    }
+
+    if (json_object_object_get_ex(conf_root, "ccov_limit", &conf_ccov_limit)){
+        this->ccov_limit=(uint64_t)json_object_get_int64(conf_ccov_limit);
+    }
+    printf("CCOV limit: %lu\n",this->ccov_limit);
+
+    int conf_modules_len=json_object_array_length(conf_modules);
+    printf("[PROCTRACER] Modules to trace: %d\n",conf_modules_len);
+    for (int i=0; i<conf_modules_len; i++){
+        char *mod_name=(char*)json_object_get_string(json_object_array_get_idx(conf_modules,i));
+        string mod_path=mod_name;
+        mod_path+=".proctracer.json";
+        symbols_t *symbols=drakvuf_get_symbols_from_rekall(mod_path.c_str());
+        for (int i=0; i < symbols->count; i++) {
```

Comment:
proctracer/proctracer.cpp:319:25: warning: comparison of integers of different signs: 'int' and 'uint64_t' (aka 'unsigned long') [-Wsign-compare]